pono.cpp

```diff
@@ -286,15 +287,21 @@ int main(int argc, char ** argv)
       FunctionalTransitionSystem fts(s);
       BTOR2Encoder btor_enc(pono_options.filename_, fts);
       const TermVec & propvec = btor_enc.propvec();
-      unsigned int num_props = propvec.size();
+      const auto & justicevec = btor_enc.justicevec();
+      unsigned int num_props =
+          pono_options.justice_ ? justicevec.size() : propvec.size();
       if (pono_options.prop_idx_ >= num_props) {
         throw PonoException(
             "Property index " + to_string(pono_options.prop_idx_)
             + " is greater than the number of properties in file "
             + pono_options.filename_ + " (" + to_string(num_props) + ")");
       }
 
-      Term prop = propvec[pono_options.prop_idx_];
+      Term prop = pono_options.justice_
+                      // This will change the transition system in place.
+                      ? LivenessToSafetyTranslator{}.translate(
+                            fts, justicevec[pono_options.prop_idx_])
+                      : propvec[pono_options.prop_idx_];
```

Comment:
We might want to make this option more configurable to accommodate future extensions.
For example, k-liveness also reduces the liveness problem to a series of safety queries.

Reply:
Added an option for this.